### PR TITLE
feat: MVP1 development tasks.

### PR DESCRIPTION
**Pull request by IRONICBo**
MVP1 Task: https://github.com/goplus/community/issues/38

### CODE REVIEW

**Comment by IRONICBo** [comment]
I will rebase commits later.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: deploy
+
+on:
+  push:
+    branches:
+      - 'feat/mvp-20240119'
```
Comment (CarlJi) [inline .github/workflows/deploy.yml]: Also need add  `dev` and `main`

File: .gitignore [hunk 2]
```diff
@@ -20,3 +20,9 @@
 
 # Go workspace file
 go.work
+
+# env file
+.env
+
+# Autogenerated files from goplus generate
+gop_autogen.go
```
Comment (CarlJi) [inline .gitignore]: If I understand correctly, the `.go` files generated by `gop` are welcome to be put into repo

File: cmd/gopcomm/community_yap.gox [hunk 3]
```diff
@@ -1,52 +1,364 @@
...
+
+get "/failed", ctx => {
+	ctx.yap "5xx", {}
+}
 
 get "/p/:id", ctx => {
+	// Get User Info
+	var userClaim *casdoorsdk.Claims
+	token, err := ctx.Request.Cookie("token")
+	if err == nil {
```
Comment (CarlJi) [inline cmd/gopcomm/community_yap.gox:58]: Why is there no reverse judgment？ like
``` go
if err != nil {}
```

File: cmd/gopcomm/community_yap.gox [hunk 4]
```diff
@@ -1,52 +1,364 @@
...
 
 var (
 	community *core.Community
+	trans *translation.Engine
 )
 
 todo := context.TODO()
+endpoint := os.Getenv("GOP_COMMUNITY_ENDPOINT")
+domain := os.Getenv("GOP_COMMUNITY_DOMAIN")
+logger, _ := zap.NewProduction()
```
Comment (CarlJi) [inline cmd/gopcomm/community_yap.gox]: handle err scenario

Comment (CarlJi) [reply cmd/gopcomm/community_yap.gox]: for log library, recommend to use https://github.com/qiniu/x/tree/main/xlog

File: cmd/gopcomm/community_yap.gox [hunk 5]
```diff
@@ -1,52 +1,364 @@
...
 	id := ctx.param("id")
 	article, _ := community.article(todo, id)
 	ctx.yap "article", {
+		"User": userClaim,
+
 		"ID":      id,
 		"Title":   article.Title,
-		"Content": article.Content,
+		"Content": article.HtmlUrl,
+		"Tags": article.Tags,
```
Comment (CarlJi) [inline cmd/gopcomm/community_yap.gox]: gop format

File: cmd/gopcomm/community_yap.gox [hunk 6]
```diff
@@ -1,52 +1,364 @@
...
+		"Cover": article.Cover,
+		"Ctime": article.Ctime,
+		// "User": article.User,
+	}
+}
+
+get "/getArticle/:id", ctx => {
+	id := ctx.param("id")
+	article, _ := community.article(todo, id)
```
Comment (CarlJi) [inline cmd/gopcomm/community_yap.gox]: handle err

File: cmd/gopcomm/community_yap.gox [hunk 7]
```diff
@@ -1,52 +1,364 @@
...
+}
+
+get "/user/:id", ctx => {
+	id := ctx.param("id")
+	// Default is the current user info
+
+	// Get User Info
+	var userClaim *casdoorsdk.Claims
+	token, err := ctx.Request.Cookie("token")
+	if err == nil {
```
Comment (CarlJi) [inline cmd/gopcomm/community_yap.gox:100]: handle `err != nil` scenario

File: cmd/gopcomm/community_yap.gox [hunk 8]
```diff
@@ -1,52 +1,364 @@
...
-		"Content": article.Content,
+		"Content": article.HtmlUrl,
+		"Tags": article.Tags,
+		"Cover": article.Cover,
+		"Ctime": article.Ctime,
+		// "User": article.User,
+	}
+}
+
+get "/getArticle/:id", ctx => {
```
Comment (CarlJi) [inline cmd/gopcomm/community_yap.gox]: it's duplicate with  `get /p/:id` ? Also, it's better to use unified style for router，

File: internal/core/community.go [hunk 9]
```diff
@@ -69,50 +89,173 @@ func New(ctx context.Context, conf *Config) (ret *Community, err error) {
...
-				Title: "Title",
-			},
-			contentSummary,
-		}
-		return
+	article = &Article{}
+	var htmlId string
+	sqlStr := "select id,title,user_id,cover,tags,content,html_id,ctime,mtime from article where id=?"
+	err = p.db.QueryRow(sqlStr, id).Scan(&article.ID, &article.Title, &article.UId, &article.Cover, &article.Tags, &article.Content, &htmlId, &article.Ctime, &article.Mtime)
+	if err == sql.ErrNoRows {
```
Comment (CarlJi) [inline internal/core/community.go:141]: when err == nil, current logic still need two check operations which seems wasteful.

suggest:
``` go
if err != nil {
    if err == ErrNotExist {
        return article, ErrNotExist 
     }
     log.Error("unexpected err", err)
      return article, err
}

```

File: internal/core/community.go [hunk 10]
```diff
@@ -69,50 +89,173 @@ func New(ctx context.Context, conf *Config) (ret *Community, err error) {
...
+		p.zlog.Warn("not found the article")
+		return article, ErrNotExist
+	} else if err != nil {
+		return article, err
+	}
+	// add author info
```
Comment (CarlJi) [inline internal/core/community.go]: recommend to remove useless code

File: internal/core/community.go [hunk 11]
```diff
@@ -69,50 +89,173 @@ func New(ctx context.Context, conf *Config) (ret *Community, err error) {
...
+	}
+	return
+}
+
+// TransHtmlUrl get translation html url
+func (p *Community) TransHtmlUrl(ctx context.Context, id string) (htmlUrl string, err error) {
+	var htmlId string
+	sqlStr := "select trans_html_id from article where id=?"
+	err = p.db.QueryRow(sqlStr, id).Scan(&htmlId)
+	if err == sql.ErrNoRows {
```
Comment (CarlJi) [inline internal/core/community.go:168]: same with above.

File: .gitignore (hunk 2 above)
Comment (IRONICBo) [reply .gitignore]: This file has caused many conflicts before.

I have removed it.

File: cmd/gopcomm/community_yap.gox (hunk 3 above)
Comment (IRONICBo) [reply cmd/gopcomm/community_yap.gox:58]: Thanks, I will update it.

File: cmd/gopcomm/community_yap.gox (hunk 4 above)
Comment (IRONICBo) [reply cmd/gopcomm/community_yap.gox]: I will update to [xlog](https://github.com/qiniu/x/tree/main/xlog).

File: cmd/gopcomm/community_yap.gox (hunk 6 above)
Comment (IRONICBo) [reply cmd/gopcomm/community_yap.gox]: @Baihhh will fix it.

**Comment by IRONICBo** [comment]
This PR have already moved to `dev`.

**Pull request closed without merging**